src/Controller/ContentPaginationController.php

```diff
@@ -0,0 +1,130 @@
+    private $router;
+
+    /** @var \App\QueryType\ContentSiblingQueryType */
+    private $contentSiblingQueryType;
+
+    /**
+     * @param \eZ\Publish\API\Repository\SearchService $searchService
+     * @param \eZ\Publish\API\Repository\ContentService $contentService
+     * @param \Symfony\Component\Routing\RouterInterface $router
+     * @param \App\QueryType\ContentSiblingQueryType $contentSiblingQueryType
+     */
+    public function __construct(
+        SearchServiceInterface $searchService,
+        ContentServiceInterface $contentService,
+        RouterInterface $router,
+        ContentSiblingQueryType $contentSiblingQueryType
+    ) {
+        $this->searchService = $searchService;
+        $this->contentService = $contentService;
+        $this->router = $router;
+        $this->contentSiblingQueryType = $contentSiblingQueryType;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function getContentPaginationAction(Request $request)
+    {
+        $contentId = (int) $request->get('contentId');
+        $content = $this->contentService->loadContentInfo($contentId);
+        $parentLocationId = (int) $request->get('parentLocationId');
+
+        $previousContent = $this->getSiblingContent($content, $parentLocationId, ContentSiblingQueryParameters::PREVIOUS_CONTENT);
+        $nextContent = $this->getSiblingContent($content, $parentLocationId, ContentSiblingQueryParameters::NEXT_CONTENT);
+
+        return $this->render('@ezdesign/parts/content_pagination.html.twig', [
+            'previous' => $previousContent ? $this->getSiblingContentURL($previousContent) : null,
+            'next' => $nextContent ? $this->getSiblingContentURL($nextContent) : null,
+        ]);
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $content
+     * @param int $parentLocationId
+     * @param string $siblingContentPosition
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Content|null
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function getSiblingContent(ContentInfo $content, int $parentLocationId, string $siblingContentPosition): ?Content
+    {
+        $queryParameters = new ContentSiblingQueryParameters([
+            'parentLocationId' => $parentLocationId,
+            'modificationDate' => $content->modificationDate,
+            'operator' => $siblingContentPosition,
+            'limit' => 1
+        ]);
+
+        $query = $this->getQuery($queryParameters);
+        $result = reset($this->searchService->findContent($query)->searchHits);
```

Comment:
I'm using reset here because I want to get first element from array without using notation as this: ```$this->searchService->findContent($query)->searchHits[0];```